staging/src/github.com/openshift/openshift-controller-manager/pkg/build/controller/build/build_controller.go

```diff
@@ -1772,32 +1831,46 @@ func (bc *BuildController) controllerConfigWork() bool {
 // `*.config.openshift.io` instances and any referenced data in the `openshift-config` namespace.
 func (bc *BuildController) handleControllerConfig() []error {
 	configErrs := bc.readClusterImageConfig()
-	err := bc.readClusterBuildControllerConfig()
+	err := bc.readAllProxyConfigs()
 	if err != nil {
 		configErrs = append(configErrs, err)
 	}
 	return configErrs
 }
 
-// readClusterBuildControllerConfig synchronizes the build controller config
-// in `builds.config.openshift.io/cluster`
-func (bc *BuildController) readClusterBuildControllerConfig() error {
+// readAllProxyConfigs synchronizes the build controller config
+// in `builds.config.openshift.io/cluster` and global proxy config in
+// in `proxies.config.openshift.io/cluster`
+func (bc *BuildController) readAllProxyConfigs() error {
 	buildConfig, err := bc.buildControllerConfigLister.Get("cluster")
-	if err != nil && !errors.IsNotFound(err) {
-		return err
-	} else if buildConfig == nil {
-		bc.setBuildDefaultProxy(nil)
-		return nil
+	if err != nil {
+		if !errors.IsNotFound(err) {
+			return err
+		}
+		// getters can return empty structs vs. nil on not founds
+		buildConfig = nil
+	}
+	globalProxy, err := bc.proxyCfgLister.Get("cluster")
+	if err != nil {
+		if !errors.IsNotFound(err) {
+			return err
+		}
+		// getters can return empty structs vs. nil on not founds
+		globalProxy = nil
 	}
 
 	if klog.V(5) {
 		configJSON, _ := json.Marshal(buildConfig)
 		if configJSON != nil {
 			klog.Infof("build controller config: %s", string(configJSON))
 		}
+		configJSON, _ = json.Marshal(globalProxy)
+		if configJSON != nil {
+			klog.Infof("global proxy config: %s", string(configJSON))
+		}
```

Comment:
I think this can fail if either is nil (previously we returned if the config values were not found).

Reply:
addressed in next push